cellfinder/core/tools/tools.py

```diff
@@ -1,19 +1,141 @@
+from functools import wraps
 from random import getrandbits, uniform
-from typing import Optional
+from typing import Callable, Optional, Type
 
 import numpy as np
+import torch
 from natsort import natsorted
 
 
-def get_max_possible_value(obj_in: np.ndarray) -> int:
+def inference_wrapper(func):
     """
-    Returns the maximum allowed value for a numpy array of integer data type.
+    Decorator that makes the decorated function/method run with
+    `torch.inference_mode` set to True.
+    """
+
+    @wraps(func)
+    def inner_function(*args, **kwargs):
+        with torch.inference_mode(True):
+            return func(*args, **kwargs)
+
+    return inner_function
```

Comment:
(You can tell I am writing docs and noticing more things as I go along :joy: )
If I read the `pytorch` docs correctly, we could use [`torch.inference_mode` as a decorator](url) directly instead of redefining it here?